### DISCUSSION TITLE
Power Button with i20230828

### DISCUSSION THREAD

**Discussion by yves2** [rern/rAudio#1413] [Bug Reports]
I was using Audiophonics Power Button successfully  with rAudio 20230123
but with rAudio 20230828  as soon as the system is up the power goes down 
it's a circle and the button is always blinking

**Comment by Alien8ions** [comment]
ATX power button is not working as should...
The power button script checks for a file which never gets created so the relevant if part of that script never gets executed..
There is also a double call for the same pins via that script and also through a dtoverlay line which gets added to config.txt in /boot.

Best choose the script way as there are 3 pins in use 
```
if [[ -e /srv/http/data/system/audiophonics ]]; then                                                
        gpio -1 mode 7 out  # soft reboot \ shutdown                                                           
        gpio -1 write 7 0                                                                           
                                                                                                    
        gpio -1 mode 15 out # boot=ok                                                             
        gpio -1 write 15 1                                                                          
                                                                                                    
        gpio -1 mode 11 in  # sw                                                                    
        gpio -1 write 11 0                                                                          
        gpio -1 wfi 11 rising                                                                       
                                                                                                    
else                                                                                                
        gpio -1 mode 7 out                                                           
        gpio -1 write 7 0                                                                           
                                                                                                    
        gpio -1 mode 15 out                                                        
        gpio -1 write 15 1                                                                          
                                                                                                    
        gpio -1 mode 11 in                                                      
        gpio -1 write 11 0                                                                          
        gpio -1 wfi 11 rising                                                                       
fi                                                                                                  
                                                                                                    
/srv/http/bash/power.sh
```

Is my rather crude way of fixing that part...

To get software reboot or shutdown
Add to power.sh:

```
if [[ -e $dirdata/system/audiophonics ]]; then                                                      
  if [[ $reboot ]]; then                                                                            
    gpio -1 mode 7 out                                                                              
    gpio -1 write 7 1                                                                               
                                                                                                    
  else                                                                                              
    gpio -1 write 7 1                                                                               
    sleep 1                                                                                         
    gpio -1 write 7 0                                                                               
  fi                                                                                                
                                                                                                    
fi

```
I have mine below the notify -blink line

've tried to get the system.sh script to work as it should but couldn't get the thing to run correctly.
At least remove the dtoverlay stuff for enabling power button

So in summary:
Pin7 signal pi->ATX for shutdown or reboot (pulse high 1 sec or leave high respectively)
Pin15 signal system finished booting pi -> ATX (so led stops blinking)
Pin11 'listen' ATX ->pi to signal button pressed (shutdown)

Hopefully this may clarify things a little...
Diego
